agent/pkg/api/main.go

```diff
@@ -134,7 +134,7 @@ func startReadingChannel(outputItems <-chan *tapApi.OutputChannelItem, extension
 		serviceMapGenerator.NewTCPEntry(mizuEntry.Source, mizuEntry.Destination, &item.Protocol)
 
 		oasGenerator := dependency.GetInstance(dependency.OasGeneratorDependency).(oas.OasGeneratorSink)
-		oasGenerator.HandleEntry(mizuEntry)
+		oasGenerator.HandleEntry(mizuEntry, &item.Protocol)
```

Comment:
Is it necessary to pass both entry and protocol? There is no protocol field on the entry?

Reply:
yeah, the change is that entry will not have protocol struct, it will only have protocol unique name

Reply:
Can't OAS get the protocol from the unique name?

Reply:
In favor of separating OAS logic from the main app, we decided to pass protocol instead of making it aware of the protocol map and the protocol unique name.